sdk/api-reference.md

```diff
@@ -6,152 +6,95 @@ icon: terminal
-  * auctionByPriceRange({ priceRange, minProceeds, maxProceeds, durationDays?, epochLength?, gamma?, tickSpacing?, numPdSlugs? })
-    * Uses `pool.tickSpacing` unless `tickSpacing` is provided here
-* withVesting({ duration?, cliffDuration? } | undefined)
-  * Omit to disable vesting. Default duration if provided but undefined is `0` for dynamic auctions.
-* withGovernance(GovernanceConfig | { useDefaults: true } | { noOp: true } | undefined)
-  * Call is required; `withGovernance()` applies standard defaults; `{ useDefaults: true }` also applies defaults; `{ noOp: true }` explicitly selects no‑op.
-* withMigration(MigrationConfig)
-* withUserAddress(address)
-* withIntegrator(address?)
-* withTime({ startTimeOffset?, blockTimestamp? } | undefined)
-  * Controls auction time reference; if omitted, factory fetches latest block timestamp and uses 30s offset
-* build(): CreateDynamicAuctionParams
-  * Ensures `gamma` finalized, fills defaults, and throws if required sections are missing
-
-Validation highlights:
-
-* token name/symbol non‑empty
-* `startTick < endTick`
-* `initialSupply > 0`, `numTokensToSell > 0`, and `numTokensToSell <= initialSupply`
-* `durationDays > 0`, `epochLength > 0`, and `durationDays * 86400` divisible by `epochLength`
-* `tickSpacing > 0`; if `gamma` provided, it must be a multiple of `tickSpacing`
-* For V4 migration config (if chosen), beneficiary percentages must sum to 10000
-
-Examples:
-
-```ts
-const params = new DynamicAuctionBuilder()
-  .tokenConfig({ name: 'My Token', symbol: 'MTK', tokenURI: 'https://example.com/mtk.json' })
-  .saleConfig({ initialSupply: parseEther('1_000_000'), numTokensToSell: parseEther('900_000'), numeraire: weth })
-  .poolConfig({ fee: 3000, tickSpacing: 60 })
-  .auctionByPriceRange({ priceRange: { startPrice: 0.0001, endPrice: 0.001 }, minProceeds: parseEther('100'), maxProceeds: parseEther('1000') })
-  .withGovernance({ useDefaults: true })
-  .withMigration({ type: 'uniswapV2' })
-  .withUserAddress(user)
-  .build()
-```
+* `tokenConfig({ name, symbol, tokenURI, yearlyMintRate? })`
+* `saleConfig({ initialSupply, numTokensToSell, numeraire })`
+* `withCurves({ numerairePrice, curves, fee?, tickSpacing?, beneficiaries? })`
+  * Requires `saleConfig()` first
+  * `curves`: Array of `{ marketCap: { start, end }, numPositions, shares }`
+  * Shares must sum to 1e18 (100%)
+* `poolConfig({ fee, tickSpacing, curves, beneficiaries? })`
+  * `curves`: Array of `{ tickLower, tickUpper, numPositions, shares }`
+* `withVesting({ duration?, cliffDuration?, recipients?, amounts? })`
+* `withGovernance({ type: 'default' | 'custom' | 'noOp' })`
+* `withMigration(MigrationConfig)`
+* `withUserAddress(address)`
+* `build()` → `CreateMulticurveParams`
+
+### Multicurve rules
+
+* First curve's `marketCap.start` = the launch price
+* Curves must be contiguous or overlapping (no gaps)
+* Shares must sum to exactly 1e18 (100%)
 
 ***
 
-### Build Results
+## Method call order
```

Comment:
Unless we specify this for the entire API interface/reference it seems out of place/context.